apps/backend/src/datasources/stfc/StfcUserDataSource.ts

```diff
@@ -306,20 +356,19 @@ export class StfcUserDataSource implements UserDataSource {
   async setUserRoles(id: number, roles: number[]): Promise<void> {
     throw new Error('Method not implemented.');
   }
-  async getRolesForUser(id: number): Promise<stfcRole[] | null> {
+  async getRolesForUser(id: number): Promise<RoleDTO[]> {
     const cachedRoles = this.stfcRolesCache.get(String(id));
     if (cachedRoles) {
       return cachedRoles;
     }
 
-    const stfcRawRolesRequest = client
-      .getRolesForUser(token, id)
-      .then((response): stfcRole[] | null => response?.return);
+    const stfcRawRolesRequest = UOWSClient.role.getRolesForUser(id.toString());
```

Comment:
Can we also catch some of the errors when calling the client and specify which endpoint of the user office client we are calling in the error message.